src/dbus/interface/source/led.rs

```diff
@@ -0,0 +1,44 @@
+use crate::input::source::iio::get_dbus_path;
+use crate::udev::device::UdevDevice;
+use std::error::Error;
+use zbus::{fdo, Connection};
+use zbus_macros::interface;
+/// DBusInterface exposing information about a HIDRaw device
+pub struct SourceLedInterface {
+    device: UdevDevice,
+}
+impl SourceLedInterface {
+    pub fn new(device: UdevDevice) -> SourceLedInterface {
+        SourceLedInterface { device }
+    }
+    /// Creates a new instance of the source hidraw interface on DBus. Returns
+    /// a structure with information about the source device.
+    pub async fn listen_on_dbus(
+        conn: Connection,
+        device: UdevDevice,
+    ) -> Result<(), Box<dyn Error>> {
+        let iface = SourceLedInterface::new(device);
+        let Ok(id) = iface.id() else {
+            return Ok(());
+        };
+        let path = get_dbus_path(id);
+        tokio::task::spawn(async move {
```

Comment:
This is done in an async function, is it *really* required just to start the object server? The `zbus` crate starts async tasks itself, see https://github.com/dbus2/zbus/blob/71834a41a008214acda27b31dc5948c8ca08a594/zbus/src/connection/mod.rs#L950

Reply:
> This is done in an async function, is it _really_ required just to start the object server? The `zbus` crate starts async tasks itself, see https://github.com/dbus2/zbus/blob/71834a41a008214acda27b31dc5948c8ca08a594/zbus/src/connection/mod.rs#L950

I copied this code from iio_imu and I mantained like that for consistency. The main problem is that right now listen_on_dbus() the function that contains the spawn is called by manager.rs in a function that must not terminate an the spawned function has a ```rust
conn.object_server().at(path.clone(), iface).await
```

so either more manager.rs logic is changed or that will have to stay like that, unless the await immediatly returns (I assume it won't, but I can try)...